osu.Game/Online/Chat/DrawableLinkCompiler.cs

```diff
@@ -39,6 +46,64 @@ private void load(OsuColour colours)
 
         protected override IEnumerable<Drawable> EffectTargets => Parts;
 
+        public MenuItem[] ContextMenuItems
+        {
+            get
+            {
+                List<MenuItem> items = new List<MenuItem>();
+
+                switch (Link.Action)
+                {
+                    case LinkAction.OpenBeatmap:
+                        int beatmapId = 0;
+
+                        if (Game?.GetBeatmapIdFromLink(Link) != null)
+                        {
+                            OsuMenuItem goToBeatmap;
+                            beatmapId = (int)Game?.GetBeatmapIdFromLink(Link);
+
+                            if (Game?.GetBeatmapFromId(beatmapId) != null)
+                                goToBeatmap = new OsuMenuItem("Go To Beatmap", MenuItemType.Highlighted, () => Game?.SelectBeatmap(beatmapId));
+                            else
+                            {
+                                goToBeatmap = new OsuMenuItem("Go To Beatmap", MenuItemType.Highlighted);
+                                goToBeatmap.Action.Disabled = true;
+                            }
+
+                            items.Add(goToBeatmap);
```

Comment:
this is quite bad:

* `GetBeatmapIdFromLink` is called twice for no reason instead of storing to a local
* `GetBeatmapFromId` shouldn't be a thing (why not get the entire beatmap by id to begin with? and if it's null then it doesn't exist locally?)
* unnecessary branch instead of doing something like
```csharp
new OsuMenuItem("Go to beatmap", MenuItemType.Highlighted)
{
    Action =
    {
        Value = () => Game?.PresentBeatmap(beatmap),
        Disabled = beatmap == null
    }
};
```

exact same review applies to the beatmap set option